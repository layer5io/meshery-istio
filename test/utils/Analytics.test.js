const nock = require('nock')

const Analytics = require('../../src/utils/Analytics')

describe('Analytics', () => {
    test('Analytics', async () => {
        const mockFn = jest.fn()
        const analytics = new Analytics({
            repo: 'all-contributors-bot',
            owner: 'all-contributors',
            user: 'mockusername',
            apiKey: 'mock api key',
            funnelId: 'mockFunnelId',
            isMock: true,
            log: {
                error: mockFn,
                info: mockFn,
            },
        })

        nock('https://api.amplitude.com')
            .post(`/httpapi`, body => {
                expect(body).toMatchSnapshot()
                return true
            })
            .reply(200)

        analytics.track('my-event')

        await analytics.finishQueue()

        expect(mockFn).not.toHaveBeenCalled()
    })
})
