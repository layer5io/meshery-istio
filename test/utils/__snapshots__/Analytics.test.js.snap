// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Analytics 1`] = `
Object {
  "api_key": "mock api key",
  "event": "[{\\"user_id\\":\\"mockusername\\",\\"event_type\\":\\"my-event\\",\\"event_properties\\":{\\"repo\\":\\"all-contributors-bot\\",\\"owner\\":\\"all-contributors\\",\\"funnel_id\\":\\"mockFunnelId\\"}}]",
}
`;
